File: domainName.py
import re
def domain_name(url):
    return re.search('(https?://)?(www\d?\.)?(?P<name>[\w-]+)\.', url).group('name')

print(domain_name('www.google.com'))


# def domain_name(url):
#     url = url.replace('www.','')
#     url = url.replace('https://','')
#     url = url.replace('http://','')
    
#     return url[0:url.find('.')]

File: domainName.js
/*
Write a function that when given a URL as a string, parses out just the domain name and returns it as a string. For example:

domainName("http://github.com/carbonfive/raygun") == "github"
domainName("http://www.zombie-bites.com") == "zombie-bites"
domainName("https://www.cnet.com") == "cnet"

*/



function domainName(url) {
    let sourceString = url.replace('http://', '').replace('https://', '').replace('www.', '').split(/[/?#]/)[0];
    let domain = sourceString.split(".")[0];
    return domain;
}

console.log(domainName("https://www.cnet.com"));

/*
return url.replace(/(https?:\/\/)?(www\.)?/, '').split('.')[0]
*/